Sources/TransactionalProperty.swift

```diff
@@ -0,0 +1,690 @@
+import Result
+
+/// A mutable, observable property that has an optionally failable action
+/// associated with the setter.
+///
+/// ## Consistency when nested
+///
+/// `TransactionalProperty` would back-propagate the validation failure if a
+/// proposed value from an outer property fails with its inner property.
+///
+/// It would also evaluate values originated from the inner property. In other
+/// words, it is possible for `value` to be invalid, causing `validations`
+/// to be a `failure`. Rely on `validations` for asserting a pass in validation.
+public final class TransactionalProperty<Value, TransactionError: Error>: ComposableMutablePropertyProtocol {
```

Comment:
> The thing that feels "weird" about PropertyEditor, though is that it behaves just like a property. So if I have a variable like emailEditor then it'd be weird to ask for emailEditor.value or emailEditor.producer.

These are more like "committed" values. But since we can't rename conformances... hmm. Though it might be more appropriate if it does not conform to `MutablePropertyProtocol` at all, given that it has different semantics than a normal mutable property.